src/common/programReplay.ts

```diff
@@ -175,6 +183,17 @@ export class ProgramReplay {
     const column = location.column;
     const sourceIndex = location.sourceIndex;
     const source = this.sourceInfo.getFullSourcePath(sourceIndex);
+
+    // TODO: Refine, initial attempt to add fileExists to locationHasAdvanced would require
+    // using vscode.workspace.fs.state or access, which are async.
+    // Making locationHasAdvanced async will require making all methods relying on it async,
+    // which will require a lot of changes.
+    // As a simpler workaround adding a 'blocked' list of regexes for paths considered synthetic
+    // in puya source maps.
+    if (PUYA_SOURCEMAPS_REGEX_BLOCKLIST.some((regex) => source.match(regex))) {
```

Comment:
See above comment on how we can resolve this for synthetic methods, however another likely source for missing locations will be third-party libraries. As when debugging the contract source might be available but not any libraries it was using.

Will investigate if there is another way to address this particular issue